### PR TITLE
Properly disconnect slaves on i2c reinitialization.

### PR DESCRIPTION
**Pull request by kareltucek**

### CODE REVIEW

File: right/src/slave_scheduler.c [hunk 1]
```diff
@@ -114,6 +116,9 @@ void InitSlaveScheduler(void)
 
     for (uint8_t i=0; i<SLAVE_COUNT; i++) {
         uhk_slave_t *currentSlave = Slaves + i;
+        if (currentSlave->isConnected && currentSlave->disconnect) {
```
Comment (mondalaci) [inline right/src/slave_scheduler.c:119]: Shouldn't all slaves be disconnected upon scheduler initialization?

Comment (mondalaci) [reply right/src/slave_scheduler.c:119]: Sorry, I guess https://github.com/UltimateHackingKeyboard/firmware/issues/399#issuecomment-1173870244 is the explanation.

Comment (kareltucek) [reply right/src/slave_scheduler.c:119]: > Sorry, I guess https://github.com/UltimateHackingKeyboard/firmware/issues/399#issuecomment-1173870244 is the explanation.

Is it?

> Shouldn't all slaves be disconnected upon scheduler initialization?

Should they? Those which are not "connected" are guaranteed to have already been properly disconnected, or never even were connected.

Comment (kareltucek) [reply right/src/slave_scheduler.c:119]: Oh, sorry, I see the misunderstanding now. Yes, we are talking watchdog-induced re-initializations.

**Comment by mondalaci** [comment]
Very well done! Stuck modules fixed.

**Pull request merged**